### ISSUE TITLE
Number labels are shown as "HOME, "CELL or "WORK on iOS if created in WebGUI

### ISSUE THREAD

**Issue by wl-lankin** [nextcloud/server#25485]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### General Information to this issue

This issue is also discussed in the nextcloud forum (https://help.nextcloud.com/t/auf-handy-steht-statt-mobil-nun-cell-usw/52601). 

Because it seems to be an issue with CardDAV it needs to be reported here and not at nextcloud/contacts.

### Steps to reproduce
1. Create contact on WebGUI (Contacts) with a phone number labeled with "Home", "Mobile" or "Work" (Sorry, in german they are called "Privat", "Mobil" and "Arbeit").
2. Sync contacts via CardDAV to iOS contacts app.
3. Open contact in contacts app on iOS. The labels will be **"HOME**, **"CELL** or **"WORK**. (All starting with an apostrophe).

### Expected behaviour
The labels should be "Privat", "Mobil" and "Arbeit" as they are called in the WebGUI and in the contacts that have been created in the iOS contacts app.

### Actual behaviour
The labels are called **"HOME**, **"CELL** or **"WORK** in iOS contacts app. (All starting with an apostrophe)

### Server configuration

**Operating system:** Ubuntu 20.04 LTS (Linux 5.4.0-65-generic x86_64)

**Web server:** Apache/2.4.41 (Ubuntu)

**Database:** maria db 10.3.25

**PHP version:** PHP 7.4.3

**Nextcloud version:** 20.0.7

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** https://nextcloud.com/install/#instructions-server (Archive File)

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.4
  - apporder: 0.11.0
  - bruteforcesettings: 2.0.1
  - calendar: 2.1.3
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.3
  - contactsinteraction: 1.1.0
  - dav: 1.16.2
  - event_update_notification: 1.2.0
  - extract: 1.3.0
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_external: 1.11.1
  - files_external_dropbox: 1.4.3
  - files_external_gdrive: 0.4.0
  - files_mindmap: 0.0.24
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.2
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - nextcloud_announcements: 1.9.0
  - notes: 4.0.2
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - onlyoffice: 6.2.0
  - password_policy: 1.10.1
  - photos: 1.2.3
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - tasks: 0.13.6
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - viewer: 1.4.0
  - workflowengine: 2.2.0
Disabled:
  - admin_audit
  - dashboard
  - encryption
  - firstrunwizard
  - recommendations
  - support
  - user_ldap
  - user_status
  - weather_status
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.7.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "appstore.experimental.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "maintenance": false,
        "filesystem_check_changes": 1,
        "mail_smtpmode": "smtp",
        "mail_smtpauth": 1,
        "mail_sendmailmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "nextant",
            "files_external_dropbox",
            "files_external_gdrive"
        ]
    }
}
```
</details>

**Are you using external storage, if yes which one:** no, internal

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** none

### Client configuration - WebGUI
**Browser:** Google Chrome Version 88.0.4324.146 (Offizieller Build) (64-Bit)

**Operating system:** Windows 10 Home 20H2

### Client configuration - iOS
**Operating system:** iOS 14.3 / iPhone 11 Pro

### Logs

Because it seems that there is not an exception or error but an issue in the data, I do not provide logs for now. If you need the logs to investigate I will try to create a clean log, without connection tries of other devices.

**Comment by eibex** [comment]
I believe this is a duplicate of https://github.com/nextcloud/contacts/issues/942

**Comment by FackelAB** [comment]
When I create or modify a contact and download this from the contact app as vcf-file I get this:

```
BEGIN:VCARD
VERSION:3.0
PRODID:-//Sabre//Sabre VObject 4.3.0//EN
UID:817a532d-3ef9-4f8f-97f0-26a8e0981195
FN:Max Mustermann
ADR;TYPE=HOME:;;Haupstraße 3;Paderborn;NRW;33161;Deutschland
EMAIL;TYPE=HOME:Test@example.com
TEL;TYPE="HOME,VOICE":08001234567890
TEL;TYPE="WORK,VOICE":09001234567890
CATEGORIES:???
ORG:Musterfirma AG
TITLE:Dr.
REV;VALUE=DATE-AND-OR-TIME:20210219T201648Z
END:VCARD
```
As written in the mentioned issue #942 in the [RFC 6350 - Section 6.4.1](https://tools.ietf.org/html/rfc6350#section-6.4.1) the line with the telephone number is correct (Value-List with quotation mark), but in https://tools.ietf.org/html/rfc6350#section-6.7.9 is the version field defined with this value: 
`The value MUST be "4.0" if the vCard corresponds to this specification.`

So I take a look in this [RFC 2426 - Section 3.6.9](https://tools.ietf.org/html/rfc2426#section-3.6.9) and here is specified, that the value-list doesn't have quotion-marks.
`TEL;TYPE=work,voice,pref,msg:+1-213-555-1234`

So I think there is a bug in cration of the vcf-structure. It must be checked if this correspond to the RFC 6350 (V 4.0) or to RFC 2426 (V3.0). 

If I change the Version-Attribute to 4.0 than the iPhone will import the Contact-Data correct.

**Comment by josabadi** [comment]
Is there any hope of a solution to the problem?

Or is this feature only used by a few people?

**Comment by rakekniven** [comment]
See german discussion in forums as well https://help.nextcloud.com/t/auf-handy-steht-statt-mobil-nun-cell-usw/52601

**Comment by JAKuhr** [comment]
This issue might be a duplicate of https://github.com/nextcloud/contacts/issues/1107 and https://github.com/nextcloud/server/issues/25485

Since this issue affects probably all ios users - and users of other popular clients like EM too -  it should be investigated with a higher priority.

Won't it be enough to set the version attribute in the vcards to "4.0" or alternatively remove the quotes from the value lists?

**Comment by eibex** [comment]
> Won't it be enough to set the version attribute in the vcards to "4.0" or alternatively remove the quotes from the value lists?

That seems what the main issue is. The reported version and actual formatting used are mismatched.

**Comment by msambg** [comment]
same problem. No solution yet?

**Comment by madfoss** [comment]
> same problem. No solution yet?

No, the nextcloud developers continue to sleep soundly and have been doing so since 2018.

**Comment by bassqualle** [comment]
I have just upgraded from NextCloud 13.x to NextCloud 20.0.2 on my personal webspace instance. To be more precise, I did a new clean install and then moved the data of two users manually by exporting the address book and re-importing it. 

After having done this the iOS devices re-synced the contacts and now I also see those strange labels on every contact card, which is quite annoying! So this issue is new to me and it is sad to see that this issue and the related ones are pending since 2018 :-/
